web/src/routes/CategoriesPage.tsx

```diff
@@ -73,8 +64,50 @@ const CategoriesPage = ({ city, pathname, cityCode, languageCode }: CityRoutePro
 
     return data
   }, [cityCode, languageCode, pathname, categoryId])
+
   const { data: parents, loading: parentsLoading, error: parentsError } = useLoadAsync(requestParents)
 
+  return {
+    categories,
+    categoriesLoading,
+    categoriesError,
+    parents,
+    parentsLoading,
+    parentsError,
+  }
+}
+
+const getBreadcrumb = (category: CategoryModel, cityName: string): BreadcrumbProps => ({
+  title: category.isRoot() ? cityName : category.title,
+  to: category.path,
+})
+
+const CategoriesPage = ({ city, pathname, cityCode, languageCode }: CityRouteProps): ReactElement | null => {
+  const previousPathname = usePreviousProp({ prop: pathname })
+  const categoryId = useParams()['*']
+  const { t } = useTranslation('layout')
+
+  const { categories, categoriesLoading, categoriesError, parents, parentsLoading, parentsError } = useCategoryData(
+    cityCode,
+    languageCode,
+    pathname,
+    categoryId,
+  )
+
+  const currentCategory = categories?.find(it => it.path === pathname)
+  useTtsPlayer(currentCategory, languageCode)
+
+  const isLeafPage = useMemo(() => {
+    if (!categories || !currentCategory) {
+      return null
+    }
+    try {
+      return new CategoriesMapModel(categories).isLeaf(currentCategory)
+    } catch {
+      return null
+    }
```

Comment:
🤔 Why do we need a try catch block here? If we don't need it, the logic can be simplified to:
``` tsx
  return categories && currentCategory? new CategoriesMapModel(categories).isLeaf(currentCategory) : null
```